packages/aws-cdk/lib/api/logs/find-cloudwatch-logs.ts

```diff
@@ -0,0 +1,101 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { Mode, SdkProvider, ISDK } from '../aws-auth';
+import { prepareSdkWithLookupRoleFor } from '../cloudformation-deployments';
+import { EvaluateCloudFormationTemplate, LazyListStackResources } from '../evaluate-cloudformation-template';
+
+// resource types that have associated CloudWatch Log Groups that should _not_ be monitored
+const IGNORE_LOGS_RESOURCE_TYPES = ['AWS::EC2::FlowLog', 'AWS::CloudTrail::Trail', 'AWS::CodeBuild::Project'];
+
+// Resource types that will create a CloudWatch log group with a specific name if one is not provided.
+// The keys are CFN resource types, and the values are the names of the physical name property of that resource.
+const RESOURCE_TYPES_WITH_IMPLICIT_LOGS: { [cfnResourceType: string]: string } = {
+  'AWS::Lambda::Function': 'FunctionName',
+};
+
+export async function findCloudWatchLogGroups(
+  sdkProvider: SdkProvider,
+  stackArtifact: cxapi.CloudFormationStackArtifact,
+): Promise<{ env: cxapi.Environment, sdk: ISDK, logGroupNames: string[] }> {
+  let sdk: ISDK;
+  const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
+  // try to assume the lookup role and fallback to the default credentials
+  try {
+    sdk = (await prepareSdkWithLookupRoleFor(sdkProvider, stackArtifact)).sdk;
+  } catch (e) {
+    sdk = (await sdkProvider.forEnvironment(resolvedEnv, Mode.ForReading)).sdk;
+  }
+
+  const listStackResources = new LazyListStackResources(sdk, stackArtifact.stackName);
+  const evaluateCfnTemplate = new EvaluateCloudFormationTemplate({
+    stackArtifact,
+    parameters: {},
+    account: resolvedEnv.account,
+    region: resolvedEnv.region,
+    partition: (await sdk.currentAccount()).partition,
+    urlSuffix: sdk.getEndpointSuffix,
```

Comment:
`urlSuffix` is of type `(region: string) => string`

Reply:
In that case it probably needs to be `sdk.getEndpointSuffix.bind(sdk)`, or alternatively `(region) => sdk.getEndpointSuffix(region)`.

It might not matter in this particular case (depends on the particular implementation of `getEndpointSuffix`), but that's just a good habit to get into.

I also don't get why `getEndpointSuffix` would be a function (since it's not a function in CFN, how will it ever be called with something other than the `region` we pass in?), but I guess that's not your doing.

Reply:
Since `urlSuffix` doesn't really need to be a method, I just updated this to be a `string` 


```
urlSuffix: sdk.getEndpointSuffix(resolvedEnv.region),
```